azkaban-common/src/main/java/azkaban/project/CronSchedule.java

```diff
@@ -18,27 +18,36 @@
 
 import com.google.common.base.Preconditions;
 import java.io.Serializable;
+import java.util.TimeZone;
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang.builder.EqualsBuilder;
 import org.apache.commons.lang.builder.HashCodeBuilder;
 
 /**
- * FlowTriggerSchedule is the logical representation of a cron-based schedule.
- * It couldn't be changed once gets constructed.
- * It will be used to schedule a trigger.
+ * FlowTriggerSchedule is the logical representation of a cron-based schedule. It couldn't be
+ * changed once gets constructed. It will be used to schedule a trigger.
  */
 public class CronSchedule implements Serializable {
 
-  private static final long serialVersionUID = -1330280892166841227L;
+  private static final long serialVersionUID = -1330280892166841228L;
```

Comment:
@acharneski  unfortunately we didn't realize that this would break deserialization for all schedules saved so far with the old UID. We need to either revert this UID change (which will break deserialization for all schedules saved with the *new* id since this change), or figure out a migration path for old schedules saved with the old UID. We are currently trying to figure it out under https://github.com/azkaban/azkaban/pull/2457.

Reply:
Sorry and thanks @li-ygerchikov , I didn't realize serialization was being used to persist objects. If that is the case, when old versions are deserialized with this new format, since there is no default constructor, I believe timeZone will be initialized as null. This will very likely cause an NPE later. I see you already reverted the ID. I'll add a PR to impute the value via an accessor to avoid any NPEs.